### ISSUE TITLE
Both JoyCon SL and SR buttons not recognized

### ISSUE THREAD

**Issue by hanglearning** [Davidobot/BetterJoy#740]
Hi, I am not sure if it is in fact BetterJoy that does not recognize the SL and SR buttons on my left and right joycon, but both yuzu and Ryujinx don't seem to recognize SL and SR when I am mapping keys, and other keys are fine. Is it just me or anyone has experienced this? Any solution? 
Thanks.

**Comment by hanglearning** [comment]
I found that as long as I "split" the joycon by clicking them on BetterJoy, then the SL and SR button is recognized in yuzu. However, another issue raises:

If I wish to play a 2 player game, and let player 1 uses the left joycon and player 2 uses the right joycon, the verticle axis on the stick is fine, but the horizontal is inverted, for both controllers. Like if I left push the stick, the stick shows going right in yuzu Configuration, and same thing for right push (shows left). This seems to be the only issue in this "split" mode.

Any solution? Thanks

**Comment by Davidobot** [comment]
This is intended behaviour... When the joycons are "joined" there aren't any buttons available to map to due to xinput limits.

You can map them to keyboard buttons using the map button.

AFAIK, the split mode stick mappings are correct, but I'll double check in an hour.

**Comment by Davidobot** [comment]
Yup, the stick mappings are correct when split. Make sure you use the Xinput device, not the original joycon.

**Comment by hanglearning** [comment]
Hi Davidobot, thank you very much for taking the time to review my issue, and more importantly, it is much more appreciated for your great efforts in making this software and that you shared it with us!

I still find my axes on the stick inverted when split. I recorded a 1 min video to indicate the problem if you could take a look (sorry for the 480p quality as that was free..) -

[https://clipchamp.com/watch/AkqC75mI5RA](url)

In my case, the left joycon's X axis inverted and the right joycon's Y axis inverted. All others worked okay. 

I've tried 
1. restarting Win10
2. re-paring the Joycon by removing them from the Bluetooth list and then paring them
3. completely deleted BetterJoy and downloaded a new copy and run
4. Reinstalled the driver (by "repairing" it) and HIDGuardian.

Yet the problem remains in yuzu (early access 1536). I couldn't test Ryujinx because for some reason the split mode did not work in Ryujinx (I can set up button mappings and Ryujinx recognized all the buttons but they did not function in games).

Maybe I can do some hack if this is a special case for my system?

Thank you very much!

**Comment by Davidobot** [comment]
Ah, I see what you mean. You want them split and upright. Not sure if this will work, but try connecting one joycon, launching BetterJoy, then clicking on the joycon icon to make it upright. Then connect your other joycon.. Not sure if that would work.

If it doesn't, let me know and I can add an option to never join joycons.

**Comment by hanglearning** [comment]
Thanks again for your timely response!
TL;DR: unfortunately did not work

Oh, I actually did not want to split them and upright. I meant to split them and use them like
![image](https://user-images.githubusercontent.com/13460425/113058298-f3124d80-917b-11eb-9974-4be30cbda528.png)
So I can play two-player games with my friend where it requires SL and SR.

But when setting up controllers and mapping buttons, yuzu and Ryujinx do not give me the option to set them like in this horizontal orientation, so no matter join mode or split mode in BetterJoy I can only set them up in an upright mode, and the issue is the inverted axis with split mode.

I tried connecting just one joycon and changed it to split mode like
![PXL_20210330_212413910 MP](https://user-images.githubusercontent.com/13460425/113059239-35885a00-917d-11eb-811d-c540b38d2d66.jpg)

And set up the controller. The axis still inverted. Then I changed it to Join mode (shown upright) in BetterJoy, but then SL and SR did not work as they intended to be.

Thank you very much! Currently, if we can just add an option to invert x or y axis for the stick, the problem shall be solved 😊
